libs/libc/machine/arch_libc.c

```diff
@@ -0,0 +1,294 @@
+/****************************************************************************
+ * libs/libc/machine/arch_libc.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <string.h>
+
+/****************************************************************************
+ * Public Functions prototypes
+ ****************************************************************************/
+
+#ifdef CONFIG_LIBC_ARCH_MEMCHR
+FAR void *arch_memchr(FAR const void *s, int c, size_t n);
```

Comment:
```suggestion
FAR void *ARCH_LIBCFUN(memchr)(FAR const void *s, int c, size_t n);
```
replace all arch define to ARCH_LIBCFUN(x) style